packages/test/test-end-to-end-tests/src/test/summaryDataStoreStats.spec.ts

```diff
@@ -172,7 +179,7 @@ describeNoCompat("Generate Summary Stats", (getTestObjectProvider) => {
         mainDataStore._root.delete("dataStore4");
         mainDataStore._root.delete("dataStore5");
 
-        sequenceNumber = await waitForSummary();
+        sequenceNumber = await waitForSummary(this.timeout() / 4);
```

Comment:
These whole tests could still time out if ensureSynchronized and waitSummaryAck always come close to the timeout values you're using. The first test can take timeout/2 * 2 + other stuff, the second test can take timeout/4 * 2 * 3 + other stuff. But it's probably true that there's a single bottleneck on failure rather than everything being slow.

Reply:
i never came up with an api i liked, but i considered writing a help that would calculate the available time left to a test base on its current duration, and the time out. something like that would be ideal, as then the timeout changes to whatever time is left minus some leeway.